src/main/java/decodex/Decodex.java

```diff
@@ -12,31 +18,43 @@
             + "| |_| | |_| |   <  __/\n"
             + "|____/ \\__,_|_|\\_\\___|\n";
 
-    /**
-     * Command Prefix Constants.
-     */
-    private static final String EXIT_COMMAND = "exit";
-
     /**
      * Decodex entry-point for the java.decodex.Decodex application.
      */
     public static void main(String[] args) {
+        DataManager dataManager = new DataManager();
+        ModuleManager moduleManager = new ModuleManager();
+        Ui ui = new Ui();
+
         System.out.println("Hello from\n" + LOGO);
 
         Scanner in = new Scanner(System.in);
-        while (true) {
+
+        boolean isExit = false;
+
+        while (!isExit) {
             System.out.print("Decodex > ");
             String userInput = in.nextLine();
 
+            // Temporary code, command functions will be moved to parser
+            Command command = null;
+
             switch (userInput) {
-            case EXIT_COMMAND:
-                System.out.print("Goodbye!");
-                System.exit(0);
+            case ExitCommand.COMMAND_WORD:
+                command = new ExitCommand();
                 break;
             default:
                 // Skeletal - Just "echos" back to us.
                 System.out.println(userInput);
             }
+
+            if (command != null) {
+                command.run(dataManager, moduleManager, ui);
+                isExit = command.isExit();
```

Comment:
Would `command instanceof ExitCommand` be a simpler way to detect the exit command? Then the `isExit()` method would not be required.